### PR TITLE
Make the deadman´s switch great again

### PR DESCRIPTION
**Pull request by Arrow768**
Now its actually useful

It triggers with absolute certainty if you let go and dont deactivate it first.
Be that because you are knocked down, stunned, shot, intentionally let go, put it in a pocket, backpack, ...

### CODE REVIEW

File: code/modules/assembly/signaler.dm [hunk 1]
```diff
@@ -60,128 +60,134 @@
...
-		if(within_jamming_range(src))
-			return
 
-		var/datum/signal/signal = new
-		signal.source = src
-		signal.encryption = code
-		signal.data["message"] = "ACTIVATE"
-		radio_connection.post_signal(src, signal)
+/obj/item/device/assembly/signaler/proc/signal()
+	log_debug("signal 1")
```
Comment (Lohikar) [inline code/modules/assembly/signaler.dm]: Debug.

File: code/modules/assembly/signaler.dm [hunk 2]
```diff
@@ -60,128 +60,134 @@
...
+	if(!radio_connection)
+		return
+	log_debug("signal 2")
```
Comment (Lohikar) [inline code/modules/assembly/signaler.dm]: Debug.

File: code/modules/assembly/signaler.dm [hunk 3]
```diff
@@ -60,128 +60,134 @@
...
+	if(within_jamming_range(src))
 		return
+	log_debug("signal 3")
```
Comment (Lohikar) [inline code/modules/assembly/signaler.dm]: Debug.

File: code/modules/assembly/signaler.dm [hunk 4]
```diff
@@ -60,128 +60,134 @@
...
+/obj/item/device/assembly/signaler/verb/deadman_it()
+	set src in usr
+	set name = "Threaten to push the button!"
+	set desc = "BOOOOM!"
+	if(!deadman)
 		deadman = 1
 		START_PROCESSING(SSprocessing, src)
 		log_and_message_admins("is threatening to trigger a signaler deadman's switch")
-		usr.visible_message("<span class='warning'>[usr] moves their finger over [src]'s signal button...</span>")
+		usr.visible_message("<span class='warning'>[usr] activates [src]'s deadman switch</span>")
```
Comment (Lohikar) [inline code/modules/assembly/signaler.dm]: Not a fan of this message. Too explicit in describing what's happening.

File: code/modules/assembly/signaler.dm [hunk 5]
```diff
@@ -60,128 +60,134 @@
...
+	else
+		deadman = 0
+		STOP_PROCESSING(SSprocessing, src)
+		usr.visible_message("<span class='warning'>[usr] deactivates [src]'s deadman switch</span>")
```
Comment (Lohikar) [inline code/modules/assembly/signaler.dm]: Same here.

File: code/modules/assembly/signaler.dm [hunk 6]
```diff
@@ -60,128 +60,134 @@
...
+	frequency = new_frequency
+	radio_connection = SSradio.add_object(src, frequency, RADIO_CHAT)
+	return
+
+/obj/item/device/assembly/signaler/process()
+	if(!deadman)
+		STOP_PROCESSING(SSprocessing, src)
+	var/mob/M = src.loc
+	if(!M || !ismob(M))
+		src.visible_message("<span class='warning'>[src]'s deadman switch is released !</span>")
```
Comment (Lohikar) [inline code/modules/assembly/signaler.dm]: Space at the end shouldn't be there, plus this phrase doesn't quite sound right. Probably too explicit again.

File: code/modules/assembly/signaler.dm (hunk 4 above)
Comment (Arrow768) [reply code/modules/assembly/signaler.dm]: What should be used instead ?

File: code/modules/assembly/signaler.dm (hunk 6 above)
Comment (Arrow768) [reply code/modules/assembly/signaler.dm]: What should be used instead ?

File: code/modules/assembly/signaler.dm (hunk 4 above)
Comment (Lohikar) [reply code/modules/assembly/signaler.dm]: Old message was fine imo. Should describe actions, not exactly what's happening.

**Comment by Arrow768** [comment]
It should be clear enough for any bystanders that a deadman switch is used so that there is no excuse they rush in, the signaler is dropped and the hostage blows up.

File: code/modules/assembly/signaler.dm (hunk 6 above)
Comment (OneOneThreeEight) [reply code/modules/assembly/signaler.dm]: src.visible_message("<span class='warning'>[src] has triggered the deadman's switch! !</span>")

Comment (Lohikar) [reply code/modules/assembly/signaler.dm]: It shouldn't explicitly state that it's a deadman switch imo. How would people know that it is from a distance?

File: code/modules/assembly/signaler.dm [hunk 7]
```diff
@@ -60,128 +60,131 @@
...
+/obj/item/device/assembly/signaler/verb/deadman_it()
+	set src in usr
+	set name = "Hold the deadman switch!"
+	set desc = "Sends a signal if dropped."
+	if(!deadman)
 		deadman = 1
 		START_PROCESSING(SSprocessing, src)
 		log_and_message_admins("is threatening to trigger a signaler deadman's switch")
-		usr.visible_message("<span class='warning'>[usr] moves their finger over [src]'s signal button...</span>")
+		usr.visible_message("<span class='warning'>[usr] presses [src]'s deadman switch...</span>")
```
Comment (skull132) [inline code/modules/assembly/signaler.dm]: Change this maybe? `presses and holds [src]'s deadman switch...`

Comment (skull132) [reply code/modules/assembly/signaler.dm]: Maybe send `usr` a little infoblurb as well, like we do with aiming. The one thing this'll do is that it'll trigger the deadman switch if the user puts the device someone into his backpack, for example.

File: code/modules/assembly/signaler.dm [hunk 8]
```diff
@@ -60,128 +60,131 @@
...
+	SSradio.remove_object(src, frequency)
+	frequency = new_frequency
+	radio_connection = SSradio.add_object(src, frequency, RADIO_CHAT)
+	return
+
+/obj/item/device/assembly/signaler/process()
+	if(!deadman)
+		STOP_PROCESSING(SSprocessing, src)
+	var/mob/M = src.loc
+	if(!M || !ismob(M))
```
Comment (skull132) [inline code/modules/assembly/signaler.dm]: One thing to check: what happens if the user puts an active deadman switch device in his pocket? Is the device's `src` still the mob? It should, ideally, still trigger the device. So you might want to check if the device is in the hands of the `loc` or not.

File: code/modules/assembly/signaler.dm [hunk 9]
```diff
@@ -60,128 +60,146 @@
...
-			STOP_PROCESSING(SSprocessing, src)
-		var/mob/M = src.loc
-		if(!M || !ismob(M))
-			if(prob(5))
-				signal()
-			deadman = 0
-			STOP_PROCESSING(SSprocessing, src)
-		else if(prob(5))
-			M.visible_message("[M]'s finger twitches a bit over [src]'s signal button!")
+	if(!SSradio)
```
Comment (Lohikar) [inline code/modules/assembly/signaler.dm]: Delete the `if SSradio` checks, this global should **always** exist.

File: code/modules/assembly/signaler.dm [hunk 10]
```diff
@@ -60,128 +60,146 @@
...
+	//If we have disabled the deadmanswitch. stop here
+	if(!deadman)
+		STOP_PROCESSING(SSprocessing, src)
+		return
+	//That there is just a fallback in case dropped is not being called
+	var/mob/M = src.loc
+	if(!M || !ismob(M))
+		deadman_trigger()
+	else if(prob(20))
+		M.visible_message("[M]'s finger twitches a bit over [src]'s deadman switch!")
```
Comment (Lohikar) [inline code/modules/assembly/signaler.dm]: Phrasing of this seems a bit off.

File: code/modules/assembly/signaler.dm [hunk 11]
```diff
@@ -60,128 +60,146 @@
...
+		deadman_trigger(user)
+
+/obj/item/device/assembly/signaler/process()
+	//If we have disabled the deadmanswitch. stop here
+	if(!deadman)
+		STOP_PROCESSING(SSprocessing, src)
+		return
+	//That there is just a fallback in case dropped is not being called
+	var/mob/M = src.loc
+	if(!M || !ismob(M))
```
Comment (Lohikar) [inline code/modules/assembly/signaler.dm]: `!M` is covered by `!ismob(M)`.

File: html/changelogs/arrow768-makethesignalergreatagain.yml [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+# Optional: Remove this file after generating master changelog.  Useful for PR changelogs that won't get used again.
+delete-after: True
+
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes: 
+  - tweak: "The deadmanswitch function from the signaler will now send a signal when the signaler is dropped."
```
Comment (Lohikar) [inline html/changelogs/arrow768-makethesignalergreatagain.yml]: deadmanswitch is not one word.

File: code/modules/assembly/signaler.dm (hunk 10 above)
Comment (Arrow768) [reply code/modules/assembly/signaler.dm]: Suggestions for something else ?

**Comment by Arrow768** [comment]
DNM until I have fixed the pocket interactions of it

Either disarm when its put into a pocket or container
Or make it trigger when its put into a pocket or container

Right now it triggers when removing it from the pocket

**Comment by Arrow768** [comment]
Pocket interactions are fixed.

**Comment by Arrow768** [comment]
Whoops. Actually its one month. But still enough time to update the review

**Comment by Arrow768** [comment]
Presented to and approved by mins.

File: code/modules/mob/living/carbon/human/inventory.dm [hunk 13]
```diff
@@ -185,7 +185,7 @@ This saves us from having to call add_fingerprint() any time something is put in
 //This is an UNSAFE proc. Use mob_can_equip() before calling this one! Or rather use equip_to_slot_if_possible() or advanced_equip_to_slot_if_possible()
 //set redraw_mob to 0 if you don't wish the hud to be updated - if you're doing it manually in your own proc.
 /mob/living/carbon/human/equip_to_slot(obj/item/W as obj, slot, redraw_mob = 1)
-
+	..()
```
Comment (skull132) [inline code/modules/mob/living/carbon/human/inventory.dm]: In theory, this calls to 
```DM
/mob/proc/equip_to_slot(obj/item/W as obj, slot)
	return
```

But I guess stack integrity is a thing.

**Pull request merged**